src/cmd.rs

```diff
@@ -16,9 +16,9 @@ pub enum Arg<D> {
 /// Represents redis commands.
 #[derive(Clone)]
 pub struct Cmd {
-    data: Vec<u8>,
+    pub(crate) data: Vec<u8>,
     // Arg::Simple contains the offset that marks the end of the argument
-    args: Vec<Arg<usize>>,
+    pub(crate) args: Vec<Arg<usize>>,
```

Comment:
Could just move the `arg_idx` method into this module and make the method `pub(crate)`